### ISSUE TITLE
TextInputLayout Material3

### ISSUE THREAD

**Issue by ArcherEmiya05** [material-components/material-components-android#2635]
Does Material3 support `TextInputLayout` I have tried it but not working correctly and looks like it still uses Material 2.

using `Widget.MaterialComponents.TextInputLayout.OutlinedBox` as a parent works with `shapeAppearance`
using `Widget.Material3.TextInputLayout.OutlinedBox` as a parent is not working with `shapeAppearance`

```
<?xml version="1.0" encoding="utf-8"?>
<resources>
    <style name="AppTextInputLayout" parent="Widget.MaterialComponents.TextInputLayout.OutlinedBox">
        <item name="materialThemeOverlay">@style/ThemeOverlay.App.TextInputLayout</item>
        <item name="shapeAppearance">@style/ShapeAppearance.App.SmallComponent</item>
    </style>

    <style name="ThemeOverlay.App.TextInputLayout" parent="">
        <!-- Outline, hint floating, and cursor color -->
        <item name="colorPrimary">@color/primaryAccent</item>
        <!-- Text color -->
        <item name="colorOnSurface">@color/primaryWhite</item>
        <item name="colorError">@android:color/holo_red_light</item>
        <item name="editTextStyle">@style/Widget.Material3.TextInputEditText.OutlinedBox</item>
    </style>
</resources>

Using it like this
```
<style name="Theme.App" parent="Theme.Material3.DayNight">
   <item name="textInputStyle">@style/AppTextInputLayout</item>
</style>
```
```

**Comment by drchen** [comment]
The theme you posted seems invalid. Can you double check if that's what you are using?

To use M3 text input layout styles, you need to change your AppTextInputLayout to inherit from Widget.Material3.TextInputLayout.OutlinedBox. To apply custom share appearance, you need to override `shapeAppearanceOverlay`, instead of `shapeAppearance`.

Let me know if this works for you.

**Comment by ArcherEmiya05** [comment]
> The theme you posted seems invalid. Can you double check if that's what you are using?
> 
> To use M3 text input layout styles, you need to change your AppTextInputLayout to inherit from Widget.Material3.TextInputLayout.OutlinedBox. To apply custom share appearance, you need to override `shapeAppearanceOverlay`, instead of `shapeAppearance`.
> 
> Let me know if this works for you.

It works now, unfortunately this is confusing as the [documentation](https://m3.material.io/components/dialogs/implementation) with Dialogs uses `shapeAppearance` as show below thus I never use the one with overlay.

![image](https://user-images.githubusercontent.com/38008900/162794409-90b4bd57-5da2-4989-9c27-0be2ab9869b5.png)

**Comment by drchen** [comment]
Right. The theme overlay is only used when rendering dialogs so ShapeAppearance is used there instead of ShapeAppearanceOverlay.

I totally agree it's a bit confusing and unfortunate we can't really align the behavior easily....
